mpact_api/mpact/telegramevents.py

```diff
@@ -40,20 +59,40 @@ def text(self):
                 chat_id=self.chat_data[ID],
                 user__id=self.from_data[ID],
             )
-        except (UserData.DoesNotExist, UserChat.DoesNotExist):
-            user_serializer = UserDataSerializer(data=self.from_data)
-            if user_serializer.is_valid():
-                # Storing user details in both UserData and UserChat Model
-                user_data_rec = UserData.objects.create(**self.from_data)
-                user_data_rec.save()
-                save_userchat(self.chat_data, user_data_rec)
-            elif user_serializer.errors[ID]:
-                # User already exists in user data model but does not exist in userchat model
-                save_userchat(self.chat_data, user_instance)
+        except UserData.DoesNotExist:
+            add_user_chat_(self.from_data, self.chat_data)
+            return True
+        except (UserChat.DoesNotExist):
+            add_user_chat_(self.from_data, self.chat_data, user_instance)
             return True
 
         # If user already exists in both the Models then just updating the
         # details(firstname, lastname) in User Data Model
         user_serializer = UserDataSerializer(user_instance, data=self.from_data)
         if user_serializer.is_valid(raise_exception=True):
             user_serializer.save()
+
+    def left_chat_participant(self):
+        # When a member leaves a chat, it makes a safe delete.
+        left_user_data = self.message[LEFT_CHAT_PARTICIPANT]
+        try:
+            user_chat_instance = UserChat.objects.get(
+                chat_id=self.chat_data[ID],
+                user__id=left_user_data[ID],
+            )
+        except UserChat.DoesNotExist as val_error:
+            raise val_error
```

Comment:
This `try` ... `except` clause raises the exception it catches. without doing anything else. Dropping the `try` block will do the same thing:
```suggestion
        user_chat_instance = UserChat.objects.get(
            chat_id=self.chat_data[ID],
            user__id=left_user_data[ID],
        )
```
However, I worry that this will raise an exception in the following scenario:
1. Bot joins group
2. Other user leaves group before they have said anything

Will the UserChat instance exist when the User leaves?

If not, we should not raise an exception, because this is a valid scenario. Maybe we just want to return? Maybe we want to create a UserChat instance with `is_deleted = True`?